alexandria/math/src/mod_arithmetics.cairo

```diff
@@ -0,0 +1,120 @@
+use option::OptionTrait;
+use traits::{Into, TryInto};
+use integer::u512;
+
+// Function that performs modular addition.
+/// # Arguments
+/// * `a` - Left hand side of addition.
+/// * `b` - Right hand side of addition.
+/// * `modulo` - modulo.
+/// # Returns
+/// * `u256` - result of modular addition
+fn add_mod(a: u256, b: u256, modulo: u256) -> u256 {
+    let mod_non_zero: NonZero<u256> = integer::u256_try_as_non_zero(modulo).unwrap();
+    let low: u256 = a.low.into() + b.low.into();
+    let high: u256 = a.high.into() + b.high.into();
+    let carry: u256 = low.high.into() + high.low.into();
+    let add_u512: u512 = u512 {
+        limb0: low.low, limb1: carry.low, limb2: carry.high + high.high, limb3: 0
+    };
+    let (_, res) = integer::u512_safe_div_rem_by_u256(add_u512, mod_non_zero);
+    res
+}
+
+// Function that return the modular multiplicative inverse. Disclaimer: this function should only be used with a prime modulo. 
+// From Fermat's little theorem, a ^ (p - 1) = 1 when p is prime and a != 0. Since a ^ (p -1 ) = a · a (p - 2) we have that 
+// a ^ (p - 2) is the multiplicative inverse of a modulo p.
```

Comment:
```suggestion
/// Function that return the modular multiplicative inverse. Disclaimer: this function should only be used with a prime modulo. 
/// From Fermat's little theorem, a ^ (p - 1) = 1 when p is prime and a != 0. Since a ^ (p -1 ) = a · a ^ (p - 2) we have that 
// a ^ (p - 2) is the multiplicative inverse of a modulo p.
```